src/frontend/js/components/SearchSuggestField/SearchSuggestField.tsx

```diff
@@ -140,125 +122,158 @@ export async function onSuggestionsFetchRequested(
+ * @param addFilter [curried] Function that adds a filter to the `CourseSearchParams` context.
+ * @param updateFullTextSearch [curried] Function that updates the query in the `CourseSearchParams` context.
  * @param event Unused: selection event.
  * @param suggestion `suggestion` as key to an anonymous object: the suggestion the user picked.
  */
-export function onSuggestionSelected(
-  this: SearchSuggestFieldBase,
+export const onSuggestionSelected = (
+  setValue: valueSetter,
+  setSuggestions: suggestionsSetter,
+  addFilter: (
+    filterName: modelName.CATEGORIES | modelName.ORGANIZATIONS,
+    payload: string,
+  ) => void,
+  updateFullTextSearch: (query: string) => void,
+) => (
   event: React.FormEvent,
   { suggestion }: { suggestion: SearchSuggestion },
-) {
+) => {
   switch (suggestion.model) {
     case modelName.COURSES:
       // Behave like a link to the course run's page
-      return location.setHref(suggestion.data.absolute_url);
+      return (location.href = suggestion.data.absolute_url);
 
     case modelName.ORGANIZATIONS:
     case modelName.CATEGORIES:
       // Update the search with the newly selected filter
-      this.props.addFilter(suggestion.model, String(suggestion.data.id));
+      addFilter(suggestion.model, String(suggestion.data.id));
       // Reset the search field state: the task has been completed
-      this.setState({
-        suggestions: [],
-        value: '',
-      });
+      setValue('');
+      setSuggestions([]);
       break;
 
     default:
       // Update the current query with the default suggestion data (the contents of the search query)
-      this.props.fullTextSearch(suggestion.data);
+      updateFullTextSearch(suggestion.data);
       break;
   }
-}
+};
 
 interface SearchSuggestFieldState {
   suggestions: SearchSuggestionSection[];
   value: string;
 }
 
+type valueSetter = React.Dispatch<
+  React.SetStateAction<SearchSuggestFieldState['value']>
+>;
+
+type suggestionsSetter = React.Dispatch<
+  React.SetStateAction<SearchSuggestFieldState['suggestions']>
+>;
+
 /**
  * Props shape for the SearchSuggestField component.
  */
 export interface SearchSuggestFieldProps {
-  addFilter: (filterName: filterGroupName, filterValue: string) => void;
-  fullTextSearch: (query: string) => void;
+  filters: APICourseSearchResponse['filters'];
 }
 
 /**
  * Non-wrapped component. Exported for testing purposes only. See SearchSuggestField.
  */
-export class SearchSuggestFieldBase extends React.Component<
-  SearchSuggestFieldProps & InjectedIntlProps,
-  SearchSuggestFieldState
-> {
-  constructor(props: SearchSuggestFieldProps & InjectedIntlProps) {
-    super(props);
-    let valueFromUrl: string;
-    try {
-      valueFromUrl = getSearchParam('query') || '';
-    } catch (e) {
-      // Do not break, default to empty
-      valueFromUrl = '';
-    }
-    this.state = { suggestions: [], value: valueFromUrl };
-  }
+export const SearchSuggestFieldBase = ({
```

Comment:
To be honnest I don't see the interest here to move it to a functional component.

Reply:
The main point was to drop these ridiculous `whatever.bind()` calls. I think removing this from all those functions makes them somewhat clearer.